### DISCUSSION TITLE
Inject Workflow Variables

### DISCUSSION THREAD

**Discussion by zhengyangyong** [argoproj/argo-workflows#6388] [Q&A]
Is there a way add a custom Workflow Variables like 'x' via coding like Java in runtime ?
I need pass variable value between pods like this :

DAG:
A -> B -> C
A : compute a variable value and save into 'x'
B & C : get 'x' and use it

Thanks & Best Regards

**Comment by tczhao** [comment]
Hi @zhengyangyong ,
I wondering if the following are what you are after
https://github.com/argoproj/argo-workflows/tree/master/examples#output-parameters
https://github.com/argoproj/argo-workflows/tree/master/examples#artifacts

**Reply by zhengyangyong** [reply]
Thanks it's helpful.
The file '/tmp/hello_world.txt' will be auto delete when DAG had completed ?

**Reply by tczhao** [reply]
The file will write to docker image writable layer by default. It will get removed once the image is gone from the machine.
